craft_application/services/provider.py

```diff
@@ -164,6 +163,32 @@ def get_provider(self, name: str | None = None) -> craft_providers.Provider:
         self._provider = self._get_provider_by_name(name)
         return self._provider
 
+    def clean_instances(self) -> None:
+        """Clean all existing managed instances related to the project."""
+        provider = self.get_provider()
+
+        current_arch = platforms.get_host_architecture()
+        build_plan = self._project.get_build_plan()
+        build_plan = [info for info in build_plan if info.build_on == current_arch]
+
+        if build_plan:
+            target = "providers" if len(build_plan) > 1 else "provider"
+            emit.progress(f"Cleaning build {target}")
```

Comment:
Do you think this would be a good time to switch to `Cleaning build environment[s]`?  I think it is more accurate and makes more sense to the end user.